synapse/swarm/syntax.py

```diff
@@ -53,11 +54,11 @@ def parse_int(text,off,trim=True):
     try:
         return int(numstr,0),off
     except Exception as e:
-        raise Exception('Expected Literal At: %d (got: %s)' % (off,text[off:off+10],))
+        raise synapse.exc.SyntaxError(expected='Literal', at=off, got=text[off:off+10])
```

Comment:
Would you be opposed to using the same kwargs as Python's SyntaxError?
https://docs.python.org/2/library/exceptions.html#exceptions.SyntaxError